tests/test_sap_success_factors/test_learner_data.py

```diff
@@ -302,3 +338,61 @@ def get_course_grade(course_id, username):
         assert learner_data[2].course_completed
         assert learner_data[2].completed_timestamp == self.NOW_TIMESTAMP
         assert learner_data[2].grade == grade
+
+    @ddt.data(
+        (True, True, 'audit', 1), (True, True, 'verified', 1),
+        (True, False, 'audit', 0), (True, False, 'verified', 1),
+        (False, True, 'audit', 0), (False, True, 'verified', 1),
+        (False, False, 'audit', 0), (False, False, 'verified', 1),
+    )
+    @ddt.unpack
+    @mock.patch('integrated_channels.integrated_channel.learner_data.EnrollmentApiClient')
+    @mock.patch('integrated_channels.integrated_channel.learner_data.GradesApiClient')
+    @mock.patch('integrated_channels.integrated_channel.learner_data.CourseApiClient')
+    def test_learner_data_audit_track_data_reporting_enabled(
+            self,
+            enable_audit_enrollment,
+            enable_reporting,
+            mode,
+            expected_data_len,
+            mock_course_api,
+            mock_grades_api,
+            mock_enrollments_api
+    ):
+        enrollment = EnterpriseCourseEnrollmentFactory(
+            enterprise_customer_user=self.enterprise_customer_user,
+            course_id=self.course_id,
+            consent_granted=True,
+        )
+
+        # Set the audit track data passback configuration
+        self.enterprise_customer.enable_audit_enrollment = enable_audit_enrollment
+        self.enterprise_customer.enable_audit_data_reporting = enable_reporting
+        self.enterprise_customer.save()
+
+        # Use self-paced course to get grades data
+        mock_course_api.return_value.get_course_details.return_value = dict(
+            pacing='self',
+            course_id=self.course_id,
+        )
+
+        # Mock grades data
+        mock_grades_api.return_value.get_course_grade.return_value = dict(
+            passed=True,
+        )
+
+        # Mock enrollment data, in particular the enrollment mode
+        mock_enrollments_api.return_value.get_course_enrollment.return_value = dict(
+            mode=mode
+        )
+
+        # Collect the learner data
+        with freeze_time(self.NOW):
+            learner_data = list(self.exporter.collect_learner_data())
+
+        assert len(learner_data) == expected_data_len
+        if self.enterprise_customer.enables_audit_data_reporting or mode == 'verified':
+            assert learner_data[0].enterprise_course_enrollment_id == enrollment.id
+            assert learner_data[0].course_id == self.course_id
+            assert learner_data[0].course_completed
+            assert learner_data[0].grade == BaseLearnerExporter.GRADE_PASSING
```

Comment:
Are you sure? This is the pattern used in all the other tests. If I change it here, maybe I should then change it in all the others?